Updates docs for smart contracts (#8)
aragon/builders-portal: docs/smart-contracts/core/plugin/PluginUUPSUpgradeable.md

## Aragon Core

###  contract `PluginUUPSUpgradeable`

An abstract contract to inherit from when creating a UUPS Upgradeable contract.

#### public variable `PLUGIN_INTERFACE_ID`

```solidity
bytes4 PLUGIN_INTERFACE_ID 
```

#### public variable `UPGRADE_PERMISSION_ID`

```solidity
bytes32 UPGRADE_PERMISSION_ID 
```

#### internal modifier `auth`

```solidity
modifier auth(bytes32 _permissionId) 
```

| Input | Type | Description |
|:----- | ---- | ----------- |
| _permissionId | bytes32 | The hash of the permission identifier |

*Auth modifier used in all components of a DAO to check the permissions.*

#### public function `supportsInterface`

> adds a IERC165 to check whether contract supports PluginUUPSUpgradeable interface or not.

Reviewer: ```suggestion
Checks whether the specified contract supports the PluginUUPSUpgradeable interface (or the IERC165 interface) or not
```

```solidity
function supportsInterface(bytes4 interfaceId) public view virtual returns (bool) 
```

| Output | Type | Description |
| ------ | ---- | ----------- |

> | [0] | bool | bool whether it supports the IERC165 or PluginUUPSUpgradeable |

Reviewer: ```suggestion
| [0] | bool | Boolean answer to whether the contract specified supports the IERC165 or the PluginUUPSUpgradeable interface|
```

*See {ERC165Upgradeable-supportsInterface}.*

#### public function `getImplementationAddress`

used to check the current base logic contract proxy delegates to.

```solidity
function getImplementationAddress() public view returns (address) 
```

| Output | Type | Description |
| ------ | ---- | ----------- |

> | [0] | address | address the address of current base logic contract. |

Reviewer: ```suggestion
| [0] | address | The address of the contract holding the current base logic. |
```
